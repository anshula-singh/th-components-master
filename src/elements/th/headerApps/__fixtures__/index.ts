import { App } from "../types";

export const apps: App[] = [
  {
    href: "/#aa",
    imageUrl: "/images/app-trailhead.svg",
    label: "Trailhead"
  },
  {
    href: "/#b",
    imageUrl: "/images/app-trailhead.svg",
    label: "Trailblazer Community"
  },
  {
    href: "/#c",
    imageUrl: "/images/app-trailhead.svg",
    label: "AppExchange"
  },
  {
    href: "/#d",
    imageUrl: "/images/app-trailhead.svg",
    label: "IdeaExchange"
  },
  {
    href: "/#e",
    imageUrl: "/images/app-trailhead.svg",
    label: "Events"
  }
];
