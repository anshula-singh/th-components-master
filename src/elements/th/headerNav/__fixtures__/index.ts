import { NavItem } from "../types";

export const items: NavItem[] = [
  {
    label: "Home",
    link: {
      href: "/home",
      target: null
    }
  },
  {
    label: "Learn",
    sections: [
      {
        label: "Learn in-demand skills",
        items: [
          {
            label: "Trails",
            link: {
              href: "/trails",
              target: null
            },
            iconUrl:
              "https://trailhead.salesforce.com/assets/navigation/trails-51e244f36ec499621c2658ee28edfd90ac8045fd7e7a06d573cc988db10db208.png",
            description: "Follow guided learning paths"
          },
          {
            label: "Trailmixes",
            link: {
              href: "/en/trailmixes",
              target: null
            },
            iconUrl:
              "https://trailhead.salesforce.com/assets/navigation/trailmixes-3f21265f15965dcee0364fb72f05131378aaccf9ecb48465c72eb0b7178a9049.png",
            description: "Create and follow custom learning playlists"
          }
        ]
      },
      {
        label: "Take a class with us",
        items: [
          {
            label: "Trailhead Academy",
            link: {
              href: "/academy",
              target: null
            },
            iconUrl:
              "https://trailhead.salesforce.com/assets/navigation/trailhead_academy-d9e26ab480d5b4b1e3f202b848326c115bd5c3f1ffbf7a18ef8065a3434fbb0d.png",
            description: "Learn directly from Salesforce experts"
          }
        ]
      },
      {
        label: "Discover your ideal career",
        items: [
          {
            label: "Career Paths",
            link: {
              href: "/career-path/",
              target: null
            },
            iconUrl:
              "https://trailhead.salesforce.com/assets/navigation/career_paths-21f06f044d4bc37debd588a04c950e027e3b97e1a6d70b9c65b076e0be12ae5d.png",
            description: "Explore which career is right for you"
          }
        ]
      },
      {
        label: "Choose your own learning adventure",
        items: [
          {
            label: "Modules",
            link: {
              href: "/modules",
              target: null
            },
            iconUrl:
              "https://trailhead.salesforce.com/assets/navigation/modules-4f688ef85de67dbc50b20093592d2eb18d6579bded02c97c195c69075a8c12f7.png",
            description: "Learn new skills in bite-sized units"
          },
          {
            label: "Projects",
            link: {
              href: "/projects",
              target: null
            },
            iconUrl:
              "https://trailhead.salesforce.com/assets/navigation/projects-203a3a08066536b68ba0db79d9f5b09732da14d0129219b7f9ff8d9699e191ad.png",
            description: "Get hands-on with step-by-step instructions"
          },
          {
            label: "Trailhead Live",
            link: {
              href: "/live/",
              target: null
            },
            iconUrl:
              "https://trailhead.salesforce.com/assets/navigation/live-ed562e5616f608e6901006f5498d3e0020b4350f03c16b5f8af051f36e90524f.png",
            description: "Watch live and on-demand videos"
          }
        ]
      }
    ],
    sectionColumns: [3, 1]
  },
  {
    label: "Credentials",
    sections: [
      {
        label: "Earn resume-worthy credentials",
        items: [
          {
            label: "Superbadges",
            link: {
              href: "/superbadges",
              target: null
            },
            iconUrl:
              "https://trailhead.salesforce.com/assets/navigation/superbadges-ed9e106ced2e77d23e404c52206fb243dd4a9bcaa7425c7bc1708bfb3ec53b84.png",
            description: "Prove your skills with real-world business challenges"
          },
          {
            label: "Certifications",
            link: {
              href: "/credentials/administratoroverview",
              target: "_self"
            },
            iconUrl:
              "https://trailhead.salesforce.com/assets/navigation/certifications-a72d70a002018ed30aa68b8606964967928ba20fc8d9e7ed9295a3f1a0b6f70c.png",
            description: "Earn role-based Salesforce credentials",
            items: [
              {
                label: "Maintain Certifications",
                link: {
                  href:
                    "/help?article=Certification-Release-Maintenance-Schedule",
                  target: null
                }
              },
              {
                label: "Verify Certifications",
                link: {
                  href: "/credentials/verification",
                  target: null
                }
              }
            ]
          }
        ]
      }
    ]
  },
  {
    label: "Community",
    sections: [
      {
        label: "Connect to opportunities",
        items: [
          {
            label: "Trailblazer Community",
            link: {
              href: "https://trailblazer.salesforce.com/about",
              target: "_blank"
            },
            iconUrl:
              "https://trailhead.salesforce.com/assets/navigation/community-b3a817ab76ef33f6cb5d0513fa37435173b38d5788169d7abd78c629fc66a706.png",
            description: "Get help, answers, and inspiration from peers"
          },
          {
            label: "Community Groups",
            link: {
              href:
                "https://trailblazercommunitygroups.com/?&utm_source=trailhead&utm_medium=web-homepage-hero&utm_campaign=community_groups",
              target: "_blank"
            },
            iconUrl:
              "https://trailhead.salesforce.com/assets/navigation/trailblazer_groups-904782d99e6104df534351fa5508bb3393a10324d5eb2ec1910add69f0386ff8.png",
            description: "Meet with local Trailblazers and learn together"
          },
          {
            label: "Events",
            link: {
              href: "/calendar",
              target: null
            },
            iconUrl:
              "https://trailhead.salesforce.com/assets/navigation/events-f635f78aac0961bb0881bd134e9a5994f26066983f08ea5961be7772a35de3f4.png",
            description: "Explore the ultimate calendar of can't-miss events"
          },
          {
            label: "Trailblazer Connect",
            link: {
              href: "/trailblazerconnect",
              target: null
            },
            iconUrl:
              "https://trailhead.salesforce.com/assets/navigation/trailblazer_connect-a9651db57f325fe5274a07594c858aebcde2beea1ff9431ff495d8d2b21cdd50.png",
            description: "Apply for a mentor and find a job"
          }
        ]
      },
      {
        label: "Get inspired",
        items: [
          {
            label: "Trailblazer Stories",
            link: {
              href: "/trailblazers",
              target: null
            },
            iconUrl:
              "https://trailhead.salesforce.com/assets/navigation/trailblazer_stories-29556249cbd350a1e9c83cd79ceb659d7cbc3ac631474e79fee3a303d34a0674.png",
            description: "Get inspired by these moving stories"
          },
          {
            label: "Trailblazer Radio",
            link: {
              href: "/podcasts",
              target: null
            },
            iconUrl:
              "https://trailhead.salesforce.com/assets/navigation/trailblazer_radio-94c704845b5167fdd2dafb03d7642af663cabc21b1fbe2ef6cb1232d475d3ed3.png",
            description: "Tune in to podcasts for tips & stories"
          }
        ]
      },
      {
        label: "Give Back",
        items: [
          {
            label: "Be a Multiplier (BAM)",
            link: {
              href: "/bam",
              target: null
            },
            iconUrl:
              "https://trailhead.salesforce.com/assets/navigation/be_a_multiplier_bam-a4fe3b838961747034b50612af950e1c1dfd4ee1a09bfef5272689d646e352cd.png",
            description: "Lead workshops for your community"
          }
        ]
      },
      {
        label: "Have fun",
        items: [
          {
            label: "Quests",
            link: {
              href: "/quests",
              target: null
            },
            iconUrl:
              "https://trailhead.salesforce.com/assets/navigation/quests-10626de96d3f68c3460d130421dd884b3b569548fa8f7f4194117a03249129f4.png",
            description: "Challenge yourself to win fun prizes"
          }
        ]
      }
    ],
    sectionColumns: [1, 3]
  },
  {
    label: "For Companies",
    sections: [
      {
        label: "Create a culture of learning",
        items: [
          {
            label: "myTrailhead",
            link: {
              href: "/mytrailhead",
              target: null
            },
            iconUrl:
              "https://trailhead.salesforce.com/assets/navigation/mytrailhead-aca929521598ad8aa5ffdb0a45040d5294c860072c5a0f2dacc4eca03e5f1cb4.png",
            description: "Reinvent learning and enablement"
          },
          {
            label: "Customer Stories",
            link: {
              href: "/customer-spotlights",
              target: null
            },
            iconUrl:
              "https://trailhead.salesforce.com/assets/navigation/customer_stories-63283ca5aa62178738755cfa8a0665dae5046be3bcc58cf9ea6f086df74c7087.png",
            description: "Learn how our customers are using Trailhead"
          },
          {
            label: "User Adoption Services",
            link: {
              href: "/user-adoption-services",
              target: null
            },
            iconUrl:
              "https://trailhead.salesforce.com/assets/navigation/user_adoption_services-f3a99fe1dd583ced34fe1872ece075d08dda9c6e2654994a0dc1670e3af12751.png",
            description: "Increase adoption and productivity"
          },
          {
            label: "Trail Tracker",
            link: {
              href:
                "https://appexchange.salesforce.com/appxListingDetail?listingId=a0N3A00000EFpAtUAL",
              target: "_blank"
            },
            iconUrl:
              "https://trailhead.salesforce.com/assets/navigation/trail_tracker-308d0dd327e11add87d273cebaf6b56f88222771754fc41be48abf679878a2e6.png",
            description: "Motivate and track employee learning"
          }
        ]
      }
    ]
  }
];
