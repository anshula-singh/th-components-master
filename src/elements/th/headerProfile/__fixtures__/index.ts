import { User } from "../types";

export const user: User = {
  avatarPath: "https://trailblazer.me/profilephoto/7291I000000PZKV/F",
  fullName: "Adam Putinski",
  badgesCount: 74,
  pointsCount: 28,
  profileUrl: {
    href: "/en/me/adamputinski"
  },
  progressRing: {
    completedPercent: 90.35
  },
  userMenu: {
    bgPath:
      "https://trailhead.salesforce.com/assets/user-menu-banner-9ab280c3acbf91c77c2e97b37a8ff6f3e16cc0cbec0bbbbb6d90f344da7c5de1.png",
    navigation: [
      {
        label: "Profile",
        link: {
          href: "https://trailblazer.me/id?lang=en_US",
          target: "_blank"
        }
      },
      {
        label: "Settings",
        link: {
          href: "https://trailblazer.me/settings?lang=en_US",
          target: "_blank"
        }
      },
      {
        label: "Favorites",
        link: {
          href: "/users/favorites"
        }
      },
      {
        label: "Hands-on Orgs",
        link: {
          href: "/users/profiles/orgs"
        }
      }
    ],
    logout: {
      label: "Log Out",
      logoutSrc: "https://trailblazer.me/services/auth/rp/oidc/logout",
      link: {
        href: "javascript:SFIDWidget.logout();"
      }
    }
  }
};
