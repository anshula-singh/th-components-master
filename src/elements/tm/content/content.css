@import "tds/reset";

* {
  border: 0;
  vertical-align: baseline;
}

article,
aside,
details,
figcaption,
figure,
footer,
header,
hgroup,
menu,
nav,
section {
  display: block;
}

ol,
ul {
  list-style: none;
}

blockquote,
q {
  quotes: none;
}

blockquote:before,
blockquote:after,
q:before,
q:after {
  content: "";
  content: none;
}

table {
  border-collapse: collapse;
  border-spacing: 0;
}

.root {
  display: flex;
  justify-content: space-between;
}

main {
  flex: 1;
  margin-right: var(--tds-spacing-32);
  max-width: 800px;
}

aside {
  font-size: var(--tds-font-size-sm);
  align-self: flex-start;
  background-color: var(--tds-color-white);
  border-radius: var(--tds-radius-lg);
  box-shadow: var(--tds-shadow-sm);
  position: sticky;
  margin-left: var(--tds-spacing-6);
  min-width: 300px;
  padding: var(--tds-spacing-5);
  top: 0;
  line-height: var(--tds-line-height-normal);
}

aside a {
  display: block;
  text-decoration: none;
  color: var(--tds-color-meteorite);
}

aside a:hover {
  display: block;
  color: var(--tds-color-brand);
}

main h2 {
  font-size: var(--tds-font-size-2xl);
  margin-bottom: var(--tds-spacing-4);
  margin-top: var(--tds-spacing-12);
  font-weight: bold;
  text-transform: capitalize;
}

main h2:first-child {
  margin-top: 0;
}

main h3,
main h4,
main h5,
main h6 {
  font-size: var(--tds-font-size-xl);
  font-weight: bold;
  margin-bottom: var(--tds-spacing-4);
  margin-top: var(--tds-spacing-10);
}

main p {
  color: var(--tds-color-black);
  margin-top: 0;
  margin-bottom: var(--tds-spacing-4);
  line-height: var(--tds-line-height-loose);
}

main p,
main li,
main dl {
  font-size: 1.0625rem; /* 17px Special */
}

main i {
  margin-right: var(--tds-spacing-2);
}

main a,
main a:visited {
  color: var(--tds-color-brand);
  text-decoration: none;
}

main a:hover {
  text-decoration: underline;
}

main strong,
b {
  font-weight: bold;
}

main em,
i {
  font-style: italic;
}

dfn {
  font-weight: bold;
  font-style: italic;
}

main pre {
  padding: var(--tds-spacing-4);
  background-color: var(--tds-color-sand);
  border-radius: var(--tds-radius-lg);
  margin-bottom: var(--tds-spacing-4);
}

main pre code,
main pre samp,
main pre kbd,
main pre var {
  font-size: var(--tds-font-size-md);
  color: var(--tds-color-meteorite);
  background-color: transparent;
  box-shadow: none;
  font-family: Monaco, Menlo, Consolas, "Courier New", monospace;
}

main code {
  font-size: var(--tds-font-size-md);
  padding: var(--tds-spacing-1);
  color: #d14;
  background-color: #f9f2f4;
  font-family: Monaco, Menlo, Consolas, "Courier New", monospace;
  border-radius: var(--tds-radius-md);
}

main samp {
  font-size: var(--tds-font-size-md);
  padding: var(--tds-spacing-1);
  background-color: #ffffff;
  font-family: Monaco, Menlo, Consolas, "Courier New", monospace;
  border-radius: var(--tds-radius-md);
  box-shadow: var(--tds-shadow-sm);
}

main var {
  font-size: var(--tds-font-size-md);
  padding: var(--tds-spacing-1);
  color: #5a0675;
  background-color: #ebe9f9;
  font-family: Monaco, Menlo, Consolas, "Courier New", monospace;
  border-radius: var(--tds-radius-md);
}

main kbd {
  font-size: var(--tds-font-size-md);
  padding: var(--tds-spacing-1);
  color: var(--tds-color-black);
  font-family: Monaco, Menlo, Consolas, "Courier New", monospace;
  border-radius: var(--tds-radius-md);
  background-color: var(--tds-color-sand);
  border: 1px solid var(--tds-color-fog);
}

main mark {
  background-color: #ffff00;
  font-weight: bold;
  padding: var(--tds-spacing-1);
  border-radius: var(--tds-radius-md);
}

main del,
main s {
  padding: var(--tds-spacing-1);
  color: #b80808;
  background-color: #ffd6d6;
  border-radius: var(--tds-radius-md);
  text-decoration: line-through;
}

main ins {
  font-weight: bold;
  padding: var(--tds-spacing-1);
  color: #067568;
  background-color: #e9f9f3;
  border-radius: var(--tds-radius-md);
  text-decoration: none;
}

main abbr,
main acronym {
  text-decoration: none;
  font-variant: none;
  border-bottom: dotted 2px var(--tds-color-ocean);
}

main abbr:hover,
main acronym:hover {
  border-bottom: solid 2px var(--tds-color-ocean);
}

main u {
  text-decoration: #d14 wavy underline;
}

main tt {
  background: linear-gradient(
    90deg,
    rgb(255, 71, 52) 0%,
    rgb(182, 53, 255) 25%,
    rgb(37, 154, 255) 50%,
    rgb(0, 216, 216) 75%,
    rgb(28, 249, 45) 100%
  );
  font-weight: bold;
  border-radius: var(--tds-radius-md);
  padding: var(--tds-spacing-1);
  color: var(--tds-color-white);
}

sup {
  vertical-align: super;
  line-height: var(--tds-line-height-none);
  font-size: var(--tds-font-size-xs);
}

sub {
  vertical-align: sub;
  line-height: var(--tds-line-height-none);
  font-size: var(--tds-font-size-xs);
}

main small {
  font-size: var(--tds-font-size-xs);
  font-style: italic;
  color: var(--tds-color-meteorite);
}

main blockquote,
main q {
  font-style: italic;
  font-size: var(--tds-font-size-xl);
  margin-top: var(--tds-spacing-6);
  line-height: var(--tds-line-height-loose);
  margin-bottom: var(--tds-spacing-6);
}

main blockquote {
  border-left: 4px solid var(--tds-color-brand);
  padding-left: var(--tds-spacing-8);
}

main q {
  position: relative;
  display: block;
  padding-left: var(--tds-spacing-10);
}

main q:before {
  content: "“";
  color: var(--tds-color-brand);
  font-size: 5rem;
  display: block;
  position: absolute;
  left: -8px;
  line-height: var(--tds-line-height-none);
  top: 30px;
}

cite {
  display: block;
  font-size: var(--tds-font-size-md);
  margin-top: var(--tds-spacing-2);
  color: var(--tds-color-meteorite);
}

main ul,
main ol,
main dl {
  line-height: var(--tds-line-height-loose);
  margin-top: var(--tds-spacing-0);
  margin-bottom: var(--tds-spacing-4);
  margin-left: var(--tds-spacing-10);
}

main ul li {
  list-style-type: disc;
  margin-top: var(--tds-spacing-6);
  margin-top: var(--tds-spacing-0);
}

main ul ul li {
  list-style-type: circle;
}

main ul ul ul li {
  list-style-type: square;
}

main ul ul ul li {
  list-style-type: square;
}

main ol li {
  list-style-type: decimal;
  margin-top: var(--tds-spacing-1);
}

main ol ol li {
  list-style-type: lower-latin;
}

main ol ol ol li {
  list-style-type: lower-roman;
}

main ol ol ol ol li {
  list-style-type: decimal;
}

main ol ol ol ol ol li {
  list-style-type: lower-latin;
}

main ol ul li {
  list-style-type: disc;
}

main ol ul ol li {
  list-style-type: lower-latin;
}

main ol ul ol ul li {
  list-style-type: square;
}

main ul ul ul ul li {
  list-style-type: none;
  margin-left: -1rem;
}

main ul ul ul ul li:before {
  content: "▫";
  padding-right: var(--tds-spacing-2);
}

main ul ul ul ul ul li {
  list-style-type: none;
  margin-left: -1rem;
}

main ul ul ul ul ul li:before {
  content: "▶";
  padding-right: var(--tds-spacing-2);
  font-size: 0.5rem;
}

main ol ul ol ul ol li {
  list-style-type: none;
  margin-left: -1rem;
}

main ol ul ol ul ol li:before {
  content: "▶";
  padding-right: var(--tds-spacing-2);
  font-size: 0.5rem;
}

main dl {
  border-left: 4px solid var(--tds-color-sand);
  margin-left: 0;
  padding-left: var(--tds-spacing-8);
}

main dt {
  font-weight: bold;
  margin-top: var(--tds-spacing-4);
  font-style: italic;
}

main table {
  background-color: var(--tds-color-white);
  border: 1px solid var(--tds-color-fog);
  margin-top: var(--tds-spacing-4);
  margin-bottom: var(--tds-spacing-4);
  width: 100%;
}

main th {
  border: 1px solid var(--tds-color-fog);
  font-weight: bold;
  padding: var(--tds-spacing-3) var(--tds-spacing-4);
  text-align: left;
  background-color: var(--tds-color-sand);
}

main td {
  border: 1px solid var(--tds-color-fog);
  padding: var(--tds-spacing-3) var(--tds-spacing-4);
}

main caption {
  font-size: var(--tds-font-size-xs);
  font-style: italic;
  color: var(--tds-color-meteorite);
  margin-bottom: var(--tds-spacing-4);
  text-align: left;
}

main img {
  max-width: 100%;
  height: auto;
  margin-top: var(--tds-spacing-4);
  margin-bottom: var(--tds-spacing-4);
  display: block;
}

main figure img {
  margin-bottom: var(--tds-spacing-4);
}

main figcaption {
  font-size: var(--tds-font-size-xs);
  font-style: italic;
  color: var(--tds-color-meteorite);
  margin-bottom: var(--tds-spacing-4);
}
