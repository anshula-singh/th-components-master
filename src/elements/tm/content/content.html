<template>
  <div class={rootClassName}>
    <main lwc:dom="manual"></main>
    <aside if:true={showHeadingsNav}>
      <template iterator:heading={headings}>
        <a
          href={heading.value.href}
          key={heading.value.id}
          onclick={onHeadingClick}
          >{heading.value.label}</a
        >
      </template>
    </aside>
  </div>
</template>
