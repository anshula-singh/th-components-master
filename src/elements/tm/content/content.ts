import { api, LightningElement, track } from "lwc";

export default class extends LightningElement {
  private observer!: MutationObserver;
  private didSetContent = false;

  @api navHide: string | null = null;

  @track private headings: {
    href: string;
    id: string;
    label: string;
  }[] = [];
  private get showHeadingsNav() {
    return this.navHide === null && this.headings.length > 0;
  }

  private get rootClassName() {
    return "root";
  }

  connectedCallback() {
    this.observer = new MutationObserver(() => {
      this.setContent();
    });
    this.observer.observe(this.template.host, {
      childList: true,
      subtree: true
    });
    this.setContent();
  }

  disconnectedCallback() {
    this.observer.disconnect();
  }

  renderedCallback() {
    if (!this.didSetContent) {
      this.didSetContent = true;
      this.setContent();
      if (window.location.hash) this.scrollToHash(window.location.hash);
    }
  }

  private onHeadingClick(e: MouseEvent) {
    let anchor = e.currentTarget as HTMLAnchorElement;
    this.scrollToHash(anchor.hash);
  }

  private scrollToHash(hash: string) {
    let heading = this.template.querySelector(hash);
    if (heading) {
      heading.scrollIntoView();
    }
  }

  private setContent() {
    let main = this.template.querySelector("main");
    if (main) {
      main.innerHTML = this.template.host.innerHTML;
      this.headings = Array.from(main.querySelectorAll("h2")).map(n => {
        let label = n.innerText.trim();
        let id = label.replace(/[^a-zA-Z0-9]/g, "_");
        let href = `#${id}`;
        // Side effects in map ftw
        n.id = id;
        return { href, id, label };
      });
    }
  }
}
