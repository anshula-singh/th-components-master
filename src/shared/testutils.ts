import { html, render, TemplateResult } from "lit-html";

export { html };

export function nextTick() {
  return new Promise(resolve => {
    process.nextTick(resolve);
  });
}

export function $(selector: string, element: HTMLElement) {
  return element.shadowRoot!.querySelectorAll(selector);
}

export function $testid(value: string, element: HTMLElement) {
  return $(`[data-testid="${value}"]`, element);
}

export function renderIntoBody(template: TemplateResult) {
  let element = document.createElement("div");
  document.body.append(element);
  render(template, element);
  return element.children[0] as HTMLElement;
}
