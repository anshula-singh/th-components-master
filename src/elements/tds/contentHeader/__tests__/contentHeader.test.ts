import { buildCustomElementConstructor } from "lwc";

import ContentHeader from "tds/contentHeader";

import { html, renderIntoBody } from "../../../../shared/testutils";
import { modul } from "../../../../../__fixtures__/content";

customElements.define(
  "tds-content-header",
  buildCustomElementConstructor(ContentHeader)
);

describe("tds-content-header", () => {
  it("renders a content header", () => {
    let element = renderIntoBody(
      html`
        <tds-content-header
          lang="ja"
          .summary=${{ ...modul, path: "/path/to/content" }}
        ></tds-content-header>
      `
    );
    let shadowRoot = element.shadowRoot!;
    expect(shadowRoot.innerHTML).toMatchHTMLSnapshot();
  });
});
