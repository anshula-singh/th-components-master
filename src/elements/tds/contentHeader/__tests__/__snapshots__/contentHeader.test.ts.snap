// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tds-content-header renders a content header: toMatchHTMLSnapshot 1`] = `
"<header style=\\"border-top-color: hsl(210,50%,42%);\\">
  <div class=\\"content-container\\">
    <tds-content-summary></tds-content-summary>
  </div>
</header>
"
`;
